packages/roosterjs-editor-dom/lib/utils/setColor.ts

```diff
@@ -34,5 +51,72 @@ export default function setColor(
                 element.dataset[dataSetName] = modeIndependentColor.lightModeColor;
             }
         }
+
+        if (isBackgroundColor && shouldAdaptTheFontColor) {
+            adaptFontColorToBackgroundColor(element, isDarkMode);
+        }
+    }
+}
+
+/**
+ * Change the font color to white or some other color, so the text can be visible with a darker background
+ * @param element The element that contains text.
+ */
+function adaptFontColorToBackgroundColor(element: HTMLElement, isDarkMode?: boolean) {
+    if (element.firstElementChild?.hasAttribute('style')) {
+        return;
+    }
+    const backgroundColor = element.style.getPropertyValue('background-color');
+
+    if (!backgroundColor) {
+        return;
+    }
+
+    const isADarkOrBrightOrNone = isADarkOrBrightColor(backgroundColor);
+    const fontColorInDarkMode = element.dataset[DarkModeDatasetNames.OriginalStyleBackgroundColor];
+    switch (isADarkOrBrightOrNone) {
+        case ColorTones.DARK:
+            element.style.color = WHITE;
+            break;
+        case ColorTones.BRIGHT:
+            //Save this value so transformColor can adjust the color when switch from dark to light mode
+            element.dataset[DarkModeDatasetNames.OriginalStyleColor] = WHITE;
+            element.style.color = fontColorInDarkMode || BLACK;
+            break;
+    }
+
+    if (!isDarkMode) {
+        delete element.dataset[DarkModeDatasetNames.OriginalStyleColor];
     }
 }
+
+function isADarkOrBrightColor(color: string): ColorTones {
+    if (color === TRANSPARENT) {
+        return ColorTones.NONE;
+    }
+
+    let lightness = calculateLightness(color);
+    if (lightness < DARK_COLORS_LIGHTNESS) {
+        return ColorTones.DARK;
+    } else if (lightness > BRIGHT_COLORS_LIGHTNESS) {
+        return ColorTones.BRIGHT;
+    }
+
+    return ColorTones.NONE;
+}
+
+/**
+ * Calculate the lightness of HSL (hue, saturation and lightness) representation
+ * @param color a RBG or RGBA COLOR
+ * @returns
+ */
+function calculateLightness(color: string) {
+    let [r, g, b] = color.match(/[\d\.]+/g) as RegExpMatchArray;
```

Comment:
why change the regex back? we should use the long regex to check if it is rgb/rgba as well